src/core/seat/keyboard.cpp

```diff
@@ -171,8 +181,8 @@ std::vector<std::function<void()>> input_manager::match_keys(uint32_t mod_state,
             /* We must be careful because the callback might be erased,
              * so force copy the callback into the lambda */
             auto callback = binding->call.key;
-            callbacks.push_back([key, callback] () {
-                (*callback) (key);
+            callbacks.push_back([key, callback, mod_binding_key] () {
```

Comment:
you can just create a `uint32_t actual_key = .. ? ...` before both loops to avoid capturing both variables and having the check `key == 0` twice.